src/multi-select/tp-multi-select.ts

```diff
@@ -342,26 +331,32 @@ export class TPMultiSelectElement extends HTMLElement {
 	highlightNextOption(): void {
 		// Get options.
 		const options: NodeListOf<TPMultiSelectOptionElement> | null = this.querySelectorAll( 'tp-multi-select-option:not([hidden="yes"])' );
+
 		if ( ! options ) {
 			this.currentlyHighlightedOption = -1;
 			return;
 		}
 
-		// Highlight next option.
-		if ( this.currentlyHighlightedOption === options.length - 1 ) {
+		// Find the next option to be highlighted.
```

Comment:
@i-am-chitti I think a more scalable solution would be:

1. If we undid all the changed code in this function
2. Add an `attributeChangedCallback` here: https://github.com/Travelopia/web-components/blob/master/src/multi-select/tp-multi-select-option.ts
3. In that function - listen for the `highlighted` attribute
4. If the option is highlighted and disabled, un-highlight it and then call this function `highlightNextOption` on the parent

This way, if the consumer accidentally highlights the component from outside this library, the library will be able to correct itself

Reply:
I think performance may degrade w.r.t this use-case -
Suppose we have a number of continuous disabled options let's say x number of continuous options are disabled. As we are looping through [all non-hidden options here](https://github.com/Travelopia/web-components/blob/master/src/multi-select/tp-multi-select.ts#L358-L364) (let's say, its count is n) in-order to find one next option, now we would need to loop through n*x number of times. 

Also, if last few options are disabled, then, we would have no way to move back to the last valid highlighted option or to say clearly, stay at the last non-disabled option (no further next navigation).

Reply:
As discussed, I've added inline-comments to improve algorithm understanding.